doc/nrf/ug_creating_matter_accessory.rst

```diff
@@ -0,0 +1,274 @@
+.. ug_creating_matter_accessory: 
+
+Creating Matter accessory
+#########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+| This tutorial shows how to create a custom device application using Matter.     
+| It will show a way to modify Matter Template sample in order to obtain a sensor application.
+
+The goal of this guide is to create a sensor, which, if active, periodically updates measured value.
+
+
+Matter Template sample
+**********************
+
+| The base for buiding a sensor device will be a Matter Template sample which provides a basic implementation of a Matter accessory.
+| First step in creating sensor device will be to copy content from :file:`samples/matter/template` to another place eg. :file:`samples/matter/sensor`.
+
+From now on, paths will be given relative to :file:`samples/matter/sensor` directory.
+
+Application Clusters
+********************
+
+Adding new functionalities to a Matter sample requires adding new application clusters.         
+
+Cluster is a representation of a single functionality within a Matter accessory.
+Each cluster contains attributes which are stored in device's memory, and commands that can be used to modify or read the state of the device, including the cluster attributes.
+
+An application can implement appropriate callback functions to be informed about specific cluster state changes. 
+They can be used to alter device's behaviour when state of a cluster is changing as a result of some external event.
+
+Editing Clusters using ZAP tool
+*******************************
+
+| Representing functionalities of an on/off switch and a sensor requires adding new clusters.
+| Adding new application clusters can be achieved by modifying ZAP file, which can be found as :file:`src/template.zap`.       
+
+
+| To edit :file:`template.zap` run the command below with ``MATTER_ROOT`` replaced by a path to the root of Matter repository:
+.. code-block:: 
+
+        $MATTER_ROOT/scripts/tools/zap/run_zaptool.sh src/template.zap 
+        
+
+| First, add a new endpoint representing a sensor device:
```

Comment:
Would be nice to add somewhere a description that "an endpoint in Matter is an addressable container for clusters appropriate for a single device type such as a sensor or a light bulb".